src/safeposix/syscalls/fs_calls.rs

```diff
@@ -2042,38 +2042,74 @@ impl Cage {
             match request {
+                //for FIONBIO, 'ptrunion' stores a pointer to an integer. If the integer is 0, the socket's
+                //nonblocking I/O is cleared. Otherwise, the socket is set for nonblocking I/O
                 FIONBIO => {
+                    //if 'ptrunion' stores a Null pointer, a 'Bad address' error is returned
+                    //otheriwse, the integer value stored in that address is returned and saved into 'arg_result'
                     let arg_result = interface::get_ioctl_int(ptrunion);
-                    //matching the tuple and passing in filedesc_enum
                     match (arg_result, filedesc_enum) {
                         (Err(arg_result), ..)=> {
-                            return arg_result; //syscall_error
+                            return arg_result;
                         }
+                        //since FIONBIO command is used with sockets, we need to make sure that the provided
+                        //file descriptor addresses a socket
+                        //otherwise, a 'Not a typewriter' error designating that the specified command
+                        //is only applicable to sockets is returned 
                         (Ok(arg_result), Socket(ref mut sockfdobj)) => {
                             let sock_tmp = sockfdobj.handle.clone();
                             let mut sockhandle = sock_tmp.write();
-
                             let flags = &mut sockfdobj.flags;
                             let arg: i32 = arg_result;
                             let mut ioctlret = 0;
-
-                            if arg == 0 { //clear non-blocking I/O
+                            //clearing nonblocking I/O on the socket if the integer is 0
+                            if arg == 0 { 
                                 *flags &= !O_NONBLOCK;
+                                //libc::fcntl is called under the hood with F_SETFL command and 0 as an argument
+                                //to set blocking I/O, and the result of the call is stored in ioctlret
+
+                                //As of now, I don't understand why besides changing the emulated flags, we perform
+                                //a real libc::fcntl syscall
                                 if let Some(ins) = &mut sockhandle.innersocket {
+
                                     ioctlret = ins.set_blocking();
                                 }
-                            } else { //set for non-blocking I/O
+                            } else {
                                 *flags |= O_NONBLOCK;
+                                //libc::fcntl is called under the hood with F_SETFL command ans O_NONBLOCK as an argument
+                                //to set nonblocking I/O, and the result of the call is stored in ioctlret
+
+                                //As of now, I don't understand why besides changing the emulated flags, we perform
```

Comment:
the innersocket is an actual kernel socket here, we need to actually set the kernel flags in this case for it to behave correctly

Reply:
Is it because there are some communication processes that lind doesn’t fully emulate, so it cannot be fully contained within lind sandbox and has to be performed by the kernel itself?

Reply:
Yeah, that actually happens a decent amount but particularly with INET sockets.